letsencrypt/cli.py

```diff
@@ -694,7 +694,7 @@ def obtain_cert(config, plugins, lineage=None):
     if config.csr is not None:
         assert lineage is None, "Did not expect a CSR with a RenewableCert"
         csr, typ = config.actual_csr
-        certr, chain = le_client.obtain_certificate_from_csr(config.domains, csr, typ)
+        certr, chain = le_client.obtain_certificate_from_csr(config.domains, csr, typ, authzr=False)
```

Comment:
Since the `authzr` argument has a default and using `--allow-subset-of-names` cannot be used with `--csr`, I think this change should be undone.